File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0]*n
        i = 0
        for num in nums:
            if num > 0:
                ans[i] = num
                i += 2

        i = 1

        for num in nums:
            if num < 0:
                ans[i] = num
                i += 2

        return ans

        

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    const ans = new Array(nums.length);

    let pos = 0;
    let neg = 1;
    for (const num of nums) {
        if (num > 0) {
            ans[pos] = num;
            pos += 2;
        } else {
            ans[neg] = num;
            neg += 2;
        }
    }

    return ans;
};